src/lib/source-handlers/azure/list-projects.ts

```diff
@@ -0,0 +1,113 @@
+import { AzureProjectData as Project } from './types';
+import base64 = require('base-64');
+import Bottleneck from 'bottleneck';
+import * as debugLib from 'debug';
+import { AzureProjectData } from './types';
+import { getAzureToken } from './get-azure-token';
+import { getAzureOrgName } from './get-azure-orgname';
+import * as needle from 'needle';
+
+const debug = debugLib('snyk:azure');
+const azureDefaultUrl = 'https://dev.azure.com';
+
+const limiter = new Bottleneck({
+  maxConcurrent: 1,
+  minTime: 500,
+});
+
+limiter.on('failed', async (error, jobInfo) => {
+  const id = jobInfo.options.id;
+  console.warn(`Job ${id} failed: ${error}`);
+  if (jobInfo.retryCount === 0) {
+    // Here we only retry once
+    console.log(`Retrying job ${id} in 25ms!`);
+    return 25;
+  }
+});
+
+async function fetchAllProjects(
+  azureDefaultUrl: string,
+  OrgName: string,
+  token: string,
+  top: number,
+): Promise<Project[]> {
+  const projectList: Project[] = [];
+  let continuationToken = 0;
+  let hasMorePages = true;
+  while (hasMorePages) {
+    try {
+      const { projects, continuationtoken, entries } = await getProjects(
+        azureDefaultUrl,
+        OrgName,
+        token,
+        top,
+        continuationToken,
+      );
+      if (continuationtoken > 0 && entries > 0) {
```

Comment:
just `if(continuationToken && entries > 0)` as we won't set it to 0 anymore

Reply:
Done